### PR TITLE
feat: Add support to Chip Widget

### LINKED ISSUE #256: feat: Add support to Chip Widget
**Description**

Add support to Flutter Chip Widget

**Requirements**

- [ ] Create Mirai Chip Parser
- [ ] Add a chip widget example 

**Additional Context**

[Add any other context or screenshots about the feature request go here.](https://api.flutter.dev/flutter/material/Chip-class.html)

### PR DESCRIPTION
**Pull request by iampranabray**
- Created Mirai Chip Parser mirai_chip_parser and mirai_chip both added

- Added a chip widget example chip_example class added where simple , rounded and colour chip added

## Related Issues

Closes #256 

## Type of Change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code refactor
- [ ] Build configuration change
- [ ] Documentation
- [ ] Chore

### CODE REVIEW

File: packages/mirai/lib/src/parsers/mirai_chip/mirai_chip_parser.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      surfaceTintColor: widget.model.surfaceTintColor.toColor(context),
+      iconTheme: widget.model.iconTheme?.parse(context),
+      padding: widget.model.padding.parse,
+      shape: widget.model.shape.parse(context),
+      autofocus: widget.model.autofocus,
+      color: MaterialStateProperty.all(widget.model.color.toColor(context)),
+      focusNode: focusNode,
+      onDeleted: () {},
+      clipBehavior: Clip.none,
+    );
```
Comment (divyanshub024) [inline packages/mirai/lib/src/parsers/mirai_chip/mirai_chip_parser.dart:64]: we missed `materialTapTargetSize`. Can we add it?

File: packages/mirai/lib/src/parsers/mirai_chip/mirai_chip_parser.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  @override
+  Widget parse(BuildContext context, MiraiChip model) {
+    return ChipWidget(
+      model: model,
+    );
+  }
+}
+
+class ChipWidget extends StatefulWidget {
```
Comment (divyanshub024) [inline packages/mirai/lib/src/parsers/mirai_chip/mirai_chip_parser.dart:25]: We don't need a stateful widget. Please add the chip widget in the parse method.

**Review by divyanshub024 — COMMENTED** [review]
Thank you so much for this PR Pranab 🤩 

I've some suggestion, please check my comments.

File: packages/mirai/lib/src/parsers/mirai_chip/mirai_chip_parser.dart [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+import 'package:mirai/mirai.dart';
+import 'package:mirai/src/parsers/mirai_icon_theme_data/mirai_icon_theme_data.dart';
+import 'package:mirai/src/parsers/mirai_rounded_rectangle_border/mirai_rounded_rectangle_border.dart';
+
```
[the remark this answers is not in the document]
Comment (iampranabray) [reply packages/mirai/lib/src/parsers/mirai_chip/mirai_chip_parser.dart:5]: okay sure!

File: packages/mirai/lib/src/parsers/mirai_chip/mirai_chip_parser.dart (hunk 1 above)
Comment (iampranabray) [reply packages/mirai/lib/src/parsers/mirai_chip/mirai_chip_parser.dart:64]: yha i will add++ it :)

File: packages/mirai/lib/src/parsers/mirai_chip/mirai_chip.dart [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    MiraiRoundedRectangleBorder? shape,
+    @Default(false) bool autofocus,
+    String? color,
+    String? backgroundColor,
+    MiraiEdgeInsets? padding,
+    double? elevation,
+    String? shadowColor,
+    String? surfaceTintColor,
+    MiraiIconThemeData? iconTheme,
+    String? materialTapTargetSize,
```
Comment (divyanshub024) [inline packages/mirai/lib/src/parsers/mirai_chip/mirai_chip.dart:33]: @iampranabray We can define it as
```
MaterialTapTargetSize? materialTapTargetSize,
```

and in the parser use it as:
```
materialTapTargetSize: model.materialTapTargetSize,
```

File: packages/mirai/lib/src/parsers/mirai_chip/mirai_chip.dart [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  factory MiraiChip.fromJson(Map<String, dynamic> json) =>
+      _$MiraiChipFromJson(json);
+
+  MaterialTapTargetSize? get materialTapTargetSizeValue {
+    return MaterialTapTargetSize.values.firstWhere(
+      (element) => element.name == materialTapTargetSize,
+      orElse: () => MaterialTapTargetSize.padded,
+    );
+  }
```
Comment (divyanshub024) [inline packages/mirai/lib/src/parsers/mirai_chip/mirai_chip.dart:44]: We can remove it.

**Review by divyanshub024 — CHANGES_REQUESTED** [review]

File: packages/mirai/lib/src/parsers/mirai_chip/mirai_chip.dart (hunk 4 above)
Comment (iampranabray) [reply packages/mirai/lib/src/parsers/mirai_chip/mirai_chip.dart:33]: ah sure . i will do that 👍

File: packages/mirai/lib/src/parsers/mirai_chip/mirai_chip.dart (hunk 5 above)
Comment (iampranabray) [reply packages/mirai/lib/src/parsers/mirai_chip/mirai_chip.dart:44]: sure

**Comment by divyanshub024** [comment]
Thank you @iampranabray for your amazing contribution. 🤩 
Let's build Mirai together 🙌🏻

**Review by divyanshub024 — APPROVED** [review]

**Pull request merged**